src/parser/Expression.ts

```diff
@@ -1290,6 +1290,90 @@ export class AnnotationExpression extends Expression {
     }
 }
 
+export class TernaryExpression extends Expression {
+    constructor(
+        readonly test: Expression,
+        readonly questionMarkToken: Token,
+        readonly consequent: Expression,
+        readonly colonToken: Token,
+        readonly alternate: Expression
+    ) {
+        super();
+        this.range = util.createRangeFromPositions(
+            test.range.start,
+            (alternate ?? colonToken ?? consequent ?? questionMarkToken ?? test).range.end
+        );
+    }
+
+    public range: Range;
+
+    transpile(state: TranspileState) {
+        let result = [];
+        let consequentInfo = util.getExpressionInfo(this.consequent);
```

Comment:
Does all this code work if `consequent` or `alternate` is `null`? It can happen for malformed expression parsed.

Reply:
No, I think it will throw a node runtime error because we call `.walk` on the expression. How do you think it should be handled?

Reply:
I updated this logic so that it will not cause a runtime error during transpile. a null consequent or alternate will be transpiled as `invalid`.